File: 单调栈/2. 下一个更大元素 I - 496/Next Greater Element I.py
#23/3/24
#https://leetcode.cn/problems/next-greater-element-i/

from collections import List 

def nextGreaterElement(nums1: List[int], nums2: List[int]) -> List[int]:

    res = [-1 for _ in range(len(nums1))]
    for i in range(len(nums1)):
        for j in range(nums2.index(nums1[i]), len(nums2)):
            if nums2[j] > nums1[i]:
                res[i] = nums2[j]
                break
    return res

File: 单调栈/2. 下一个更大元素 I - 496/Next Greater Element I.cpp
/*
    23/03/24
    https://leetcode.cn/problems/next-greater-element-i/
 */

#include <iostream>
#include <vector>

using namespace std;

/*
    solution: 暴力
    result: ✅
    TC: O(n * m)
    SC: O(1)
 */

vector<int> nextGreaterElement1(vector<int>& num1, vector<int>& num2) {

    int lengthOfNum1 = num1.size();
    int lengthOfNum2 = num2.size();
    vector<int> result(lengthOfNum1, -1);

    for (int i = 0; i < lengthOfNum1; i++) {
        int n1 = num1[i];
        vector<int>::iterator tmpIndexInNum2 = find(num2.begin(), num2.end(), n1);
        if (tmpIndexInNum2 != num2.end()) {
            for (int j = distance(num2.begin(), tmpIndexInNum2) + 1; j < lengthOfNum2; j++) {
                if (num2[j] > n1) {
                    result[i] = num2[j];
                    break;
                }
            }
        }
    }

    return result;
}